DiffMatchPatch.zig

```diff
@@ -2,9 +2,32 @@ const DiffMatchPatch = @This();
 
 const std = @import("std");
 const testing = std.testing;
+const Allocator = std.mem.Allocator;
 const ArrayListUnmanaged = std.ArrayListUnmanaged;
 const DiffList = ArrayListUnmanaged(Diff);
 
+/// Deinit an `ArrayListUnmanaged(Diff)` and the allocated slices of
+/// text in each `Diff`.
+pub fn deinitDiffList(allocator: Allocator, diffs: *DiffList) void {
```

Comment:
nit: this function should probably be moved after the `diff` function. The `diff` function should also document that the return value needs to be de allocated with this function.

Reply:
I'll do that. I have some ideas about making the interface more Zig-ish but that can wait.

Reply:
So, thought here: should the `DiffList` be moved under `diff` as well, then? Because it's good to keep functions next to the structure they apply to